### PR TITLE
chore: Change nginx docker image to the official one with geoip2 support

### PR DESCRIPTION
**Pull request by 7mochi**
# Describe your changes
Replace the nginx image with the official version with geoip2 module support. To add third-party modules to the official nginx image I followed the [instructions specified in its official repository](https://github.com/nginxinc/docker-nginx/blob/master/modules/README.md)

I feel that this change is beneficial because this new image has more functionalities that the other image didn't have such as [using environment variables in the nginx config](https://github.com/docker-library/docs/tree/master/nginx#using-environment-variables-in-nginx-configuration-new-in-119) which I also plan to implement if this pr is accepted, that will eliminate the [step of editing the `nginx.conf` file (using docker setup, not the manual one)](https://github.com/osuAkatsuki/bancho.py/wiki/4.-Setting-Up-(Docker)#configuring-a-reverse-proxy-well-use-nginx), making the installation easier

## Checklist
- [x] The changes pass pre-commit checks (`make lint`)
- [x] The changes follow coding style

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -62,6 +62,11 @@ AUTOMATICALLY_REPORT_PROBLEMS=False
 SSL_CERT_PATH=/home/user/certs/fullchain.crt
 SSL_KEY_PATH=/home/user/certs/private.key
 
+# XXX: Uncomment this if you have downloaded the database from maxmind.
+# Change the path to the .mmdb file you downloaded
```
Comment (cmyui) [inline .env.example:66]:
```suggestion
# Change the path to the .mmdb file you downloaded, uncomment here and in docker-compose.yml
```

Comment (cmyui) [reply .env.example:66]: maybe useful

**Comment by cmyui** [comment]
is this tested?

**Comment by 7mochi** [comment]
> is this tested?

I tested it and it works, but it would be great if someone else could test it as well

File: docker/nginx-with-geoip2/Dockerfile
Comment (cmyui) [inline docker/nginx-with-geoip2/Dockerfile:1]: ur skillset is wild if u wrote this

**Review by tsunyoku — APPROVED** [review]
tested and works, very nice

**Comment by tsunyoku** [comment]
hf fixing the merge conflicts @cmyui :D

**Comment by NiceAesth** [comment]
@7mochi ...aaaaaaaaand BOOM

**Pull request merged**